robot_smach_states/src/robot_smach_states/world_model.py

```diff
@@ -224,35 +228,52 @@ class CheckEmpty(smach.State):
     """
     Check whether a volume of an entity is filled
     """
-    def __init__(self, robot, segmented_entity_ids_designator, entity_designator, volume, threshold=None):
+    def __init__(self, robot, segmented_entity_ids_designator, entity_designator, volume, threshold_perc=None,
```

Comment:
I don't think this is a good way of doing this. In my opinion it would be better to make two tests in the challenge itself. One for occupied and have that one fall back into one to determine partially occupied. That way this function can simply remain a volume occupancy check

Reply:
This state is currently combining three checks:
1: are there any entities in the volume
2: is the sum of entity volumes in the volume higher than a set volume [m3]
3: is the sum of entity volumes in the volume higher than a set percentage of the volume.
I think we should try to separate these three in three different checks.

Reply:
This state has been split